dspace-api/src/main/java/org/dspace/app/sherpa/v2/SHERPAResponse.java

```diff
@@ -0,0 +1,533 @@
+
+                // To determine which option # we are, inspect article versions and set
+                allowed.add(permittedVersion.getArticleVersion());
+                if ("submitted".equals(permittedVersion.getArticleVersion())) {
+                    submittedOption++;
+                    currentOption = submittedOption;
+                } else if ("accepted".equals(permittedVersion.getArticleVersion())) {
+                    acceptedOption++;
+                    currentOption = acceptedOption;
+                } else if ("published".equals(permittedVersion.getArticleVersion())) {
+                    publishedOption++;
+                    currentOption = publishedOption;
+                }
+                permittedVersion.setOption(currentOption);
+                permittedVersions.add(permittedVersion);
+
+                // Populate the old indicators into the publisher policy object
+                if (allowed.contains("submitted")) {
+                    sherpaPublisherPolicy.setPreArchiving("can");
+                }
+                if (allowed.contains("accepted")) {
+                    sherpaPublisherPolicy.setPostArchiving("can");
+                }
+                if (allowed.contains("published")) {
+                    sherpaPublisherPolicy.setPubArchiving("can");
+                }
+
+            }
+            sherpaPublisherPolicy.setPermittedVersions(permittedVersions);
+        }
+
+        return sherpaPublisherPolicy;
+    }
+
+    /**
+     * Parse permitted version JSON and populate new bean from the data
+     * @param permitted - each 'permitted_oa' node in the JSON array
+     * @return populated SHERPAPermittedVersion object
+     */
+    private SHERPAPermittedVersion parsePermittedVersion(JSONObject permitted) {
+
+        SHERPAPermittedVersion permittedVersion = new SHERPAPermittedVersion();
+
+        // Get the article version, which is ultimately used for the ticks / crosses
+        // in the UI display. My assumptions around translation:
+        // submitted = preprint
+        // accepted = postprint
+        // published = pdfversion
+        String articleVersion = "unknown";
+        String versionLabel = "Unknown";
+
+        if (permitted.has("article_version")) {
+            JSONArray versions = permitted.getJSONArray("article_version");
+            articleVersion = versions.getString(0);
+            permittedVersion.setArticleVersion(articleVersion);
+            log.debug("Added allowed version: " + articleVersion + " to list");
+        }
+
+        if ("submitted".equals(articleVersion)) {
+            versionLabel = I18nUtil.getMessage("jsp.sherpa.submitted-version-label");
+        } else if ("accepted".equals(articleVersion)) {
+            versionLabel = I18nUtil.getMessage("jsp.sherpa.accepted-version-label");
+        } else if ("published".equals(articleVersion)) {
+            versionLabel = I18nUtil.getMessage("jsp.sherpa.published-version-label");
+        }
```

Comment:
This is a good point. Since the key ('submitted' or 'accepted' or 'published') is already stored in the articleVersion member variable and is available via a public getter as was the label, I have simply removed the versionLabel and its getters/setters to avoid any confusion.
The key can be used to build an i18n string.
See commit: 7cb1d1fa